### ISSUE TITLE
Background auto join doesnt seem to be working

### ISSUE THREAD

**Issue by Enissay** [ge-ku/AutoJoin-for-SteamGifts#71]
As the title says, it used to work months ago, but since I have to manually click the button to do it, then come back later to do the same :<

Running:
- Win10 x64
- Chrome Version 129.0.6668.100 (Official Build) (64-bit)

Here's my config:
![image](https://github.com/user-attachments/assets/36e19f72-55ff-4272-9cbe-78094377136a)

**Comment by ge-ku** [comment]
Please check if there are any errors when you go to extension page (Extensions > Manage Extensions > Details or paste chrome://extensions/?id=bchhlccjhoedhhegglilngpbnldfcidc into address bar). Enable Developer mode in top right corner.

You'll then see:
Inspect views
service worker
html/offscreen.html

Click on both of these and if you see errors or some kind of logs, send them.

**Comment by banjorae** [comment]
neither is working for me for months, keeping the page open or background it just won't join anything, have been using it for years its due recent updates that stopped working

**Comment by ge-ku** [comment]
> neither is working for me for months, keeping the page open or background it just won't join anything, have been using it for years its due recent updates that stopped working

Recent updates were made because Chrome dropped manifest v2 extension support, so it had to be updated.
Check reply above for logs you can send, perhaps we'll be able to fix it (as it works for me as it is right now, I can't reproduce your issue).

**Comment by ge-ku** [comment]
> @ge-ku I remember than uBlock Origin dev said the following: **_"ABP 4.1 uses such trick to keep its service worker alive and ready to act: the service worker is force-restarted each time it is terminated by the browser, so every 30 seconds when the browser is idle."_**
> 
> Maybe you need to implement a similar trick for your extension to keep the service worker alive? 🤔

This shouldn't be an issue as it uses [alarms API](https://developer.chrome.com/docs/extensions/reference/api/alarms) and doesn't store anything in memory that is needed next time alarm is fired. And alarms successfuly work on my end (and yes, it becomes "inactive" between runs).



> Please check if there are any errors when you go to extension page (Extensions > Manage Extensions > Details or paste chrome://extensions/?id=bchhlccjhoedhhegglilngpbnldfcidc into address bar). Enable Developer mode in top right corner.
> 
> You'll then see: Inspect views service worker html/offscreen.html
> 
> Click on both of these and if you see errors or some kind of logs, send them.

Can you try this and see if there's anything logged?

**Comment by ge-ku** [comment]
> @ge-ku Do I have to keep that window open?! It started to work...

Nope, it should work the same with or without the window, strange.

**Comment by ge-ku** [comment]
@sn-o-w thanks! Sorry for the late reply, I don't have access to my usual computer right now. I'll take a closer look at your PR later but at first glance I'm not a fan of sending message every 20 seconds to be honest. Maybe we could use `chrome.runtime.onStartup` for creating alarms instead?

**Comment by sn-o-w** [comment]
@ge-ku I have updated my own modified version of the extension that keeps the service worker alive, and it seems like something is wrong now:

![Image](https://github.com/user-attachments/assets/623ffa65-4541-4fcd-8367-753d3585ef12)

Line 864, at least in my modified version of the extension, is the following:  
`chrome.tabs.sendMessage(sender.tab.id, { granted: 'true' });`

Do you also encounter this error, or am I the only one experiencing it? 🤔

**Comment by sn-o-w** [comment]
@ge-ku Oh, nevermind... I no longer see the error and the extension works now. 👍
